arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedReaderBuilder.java

```diff
@@ -70,11 +70,28 @@ protected VectorizedReaderBuilder(
       Map<Integer, ?> idToConstant,
       Function<List<VectorizedReader<?>>, VectorizedReader<?>> readerFactory,
       BiFunction<org.apache.iceberg.types.Type, Object, Object> convert) {
+    this(
+        expectedSchema,
+        parquetSchema,
+        setArrowValidityVector,
+        idToConstant,
+        readerFactory,
+        convert,
+        ArrowAllocation.rootAllocator());
+  }
+
+  protected VectorizedReaderBuilder(
```

Comment:
We add a new constructor which let's us specify the allocator used for the ReaderBuilder's children. This lets us do testing without relying on synchronization on the allocator. Each test now uses it's own child allocator which we can track independently

Reply:
nit: annotate this with `@VisibleForTesting`?